DataFormats/VertexReco/interface/TrackTimeLifeInfo.h

```diff
@@ -0,0 +1,73 @@
+#ifndef DataFormats_VertexReco_TrackTimeLifeInfo_h
+#define DataFormats_VertexReco_TrackTimeLifeInfo_h
+
+/**
+  \class    TrackTimeLifeInfo
+  \brief    Structure to hold time-life information
+
+  \author   Michal Bluj, NCBJ, Warsaw
+*/
+
+#include "DataFormats/GeometryCommonDetAlgo/interface/GlobalError.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+class TrackTimeLifeInfo {
+public:
+  TrackTimeLifeInfo();
+  ~TrackTimeLifeInfo() {}
+
+  // Secondary vertex
+  void setSV(reco::Vertex sv) {
+    sv_ = sv;
+    hasSV_ = true;
+  }
+  const reco::Vertex& sv() const { return sv_; }
+  bool hasSV() const { return hasSV_; }
+  void setFlightVector(GlobalVector flight_vec, const GlobalError& flight_cov) {
+    flight_vec_ = flight_vec;
+    flight_cov_ = flight_cov;
+  }
+  // Flight-path
+  const GlobalVector& flightVector() const { return flight_vec_; }
+  const GlobalError& flightCovariance() const { return flight_cov_; }
+  void setFlightLength(Measurement1D flightLength) { flightLength_ = flightLength; }
+  const Measurement1D& flightLength() const { return flightLength_; }
+  // Point of closest approach
+  void setPCA(GlobalPoint pca, const GlobalError& pca_cov) {
+    pca_ = pca;
+    pca_cov_ = pca_cov;
+  }
+  const GlobalPoint& pca() const { return pca_; }
+  const GlobalError& pcaCovariance() const { return pca_cov_; }
+  // Impact parameter
+  void setIP(GlobalVector ip_vec, const GlobalError& ip_cov) {
+    ip_vec_ = ip_vec;
+    ip_cov_ = ip_cov;
+  }
+  const GlobalVector& ipVector() const { return ip_vec_; }
+  const GlobalError& ipCovariance() const { return ip_cov_; }
+  void setIPLength(Measurement1D ipLength) { ipLength_ = ipLength; }
+  const Measurement1D& ipLength() const { return ipLength_; }
+  // Track
+  void setTrack(const reco::Track* track) { track_ = track; }
+  const reco::Track* track() const { return track_; }
+  bool hasTrack() const { return track_ != nullptr; }
+  void setBFiled_z(float bField_z) { bField_z_ = bField_z; }
```

Comment:
please fix the typo `setBFiled_z` to `setBField_x`

Reply:
It is consistently `z`, do I miss something?

Reply:
the method is called `setBFiled_z` (which looks like a type). I guess `setBField_z` was meant instead.

Reply:
Thanks, now I got it!